pkg/controller/datavolume/pvc-clone-controller.go

```diff
@@ -302,23 +358,6 @@ func (r *PvcCloneReconciler) sourceInUse(dv *cdiv1.DataVolume, eventReason strin
 	return len(pods) > 0, nil
 }
 
-func (r *PvcCloneReconciler) cleanup(syncState *dvSyncState) error {
-	dv := syncState.dvMutated
-
-	// This cleanup should be done if dv is marked for deletion or in case it succeeded
-	if dv.DeletionTimestamp == nil && dv.Status.Phase != cdiv1.Succeeded {
-		return nil
-	}
-
-	r.log.V(3).Info("Cleanup initiated in dv PVC clone controller")
-
-	if err := r.populateSourceIfSourceRef(dv); err != nil {
```

Comment:
I find the semantics of the `cleanup` callback pretty weird since https://github.com/kubevirt/containerized-data-importer/pull/2722/commits/5689e10f313a91d2d49ff99a1fe7c40d5bf7d804

It is basically called at each reconcile and the callback has to actually figure out "should I really cleanup now or not".  I avoided it as long as I could.

Reply:
same as the prepare callback is being called every reconcile and you dont need to prepare all the time.. and thats why people put in the prepare a call for the cleanup which was wrong.

Reply:
I honestly didn't have an issue with prepare calling cleanup.  Nevertheless, no reason to go back and forth on this, I begrudgingly re-added a cleanup callback in a commit that you maybe haven't got to yet